cryptoki/src/functions/slot_token_management.rs

```diff
@@ -108,6 +109,50 @@ impl Pkcs11 {
             Ok(TokenInfo::new(token_info))
         }
     }
+
+    /// Get all mechanisms support by a slot
+    pub fn get_mechanisms(&self, slot: Slot) -> Result<Vec<MechanismType>> {
+        let mut mechanism_count = 0;
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetMechanismList)(
+                slot.into(),
+                std::ptr::null_mut(),
+                &mut mechanism_count,
+            ))
+            .into_result()?;
+        }
+
+        let mut mechanisms = vec![0; mechanism_count.try_into()?];
+
+        unsafe {
+            Rv::from(get_pkcs11!(self, C_GetMechanismList)(
+                slot.into(),
+                mechanisms.as_mut_ptr(),
+                &mut mechanism_count,
+            ))
+            .into_result()?;
+        }
+
```

Comment:
That sounds very reasonable to me, I didn't get what it was doing before, thank you for the explanation.
But I'm still a bit confused about the handling of this, what if the size increased? Shouldn't I make an if-check so that it only `resize`s if it's smaller? The other's don't have that, they just default-insert a `Slot::new(0)`, isn't that a problem?

Reply:
If what I said is correct, I believe [`truncate`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.truncate) would be the more appropriate solution here, no additional if-check required.

Reply:
In the meanwhile I read a bit more about it, if it's too small, truncate seems to make sense here, but if it's too big, it will actually throw an error that isn't very useful.
Maybe we should consider putting this into a loop to make sure that if it's too big we adjust the size? In any case, I will make a separate PR for this, for now I believe `truncate` is good enough?

Reply:
Yes, if the size increased it should return `CKR_BUFFER_TOO_SMALL` which is not really useful indeed. I think `truncate` is definitely good enough for now. Even if a not-very-helpful error is returned, this function will stay safe. I am not sure if it's worth it to work on until we see that this is definitely a problem actually 😬

Maybe that can happen in some weird race condition? I would say it's good enough to keep `truncate` everywhere and open an issue about it in case anyone hits that problem one day.

Reply:
Alright, I guess this is resolved then.

Reply:
See #34.